### PR TITLE
feat(InputGroup): Add InputGroup to PF4

### PR DESCRIPTION
**Pull request by boaz0**
**What**:

Added Patternfly 4 input group to react.

### CODE REVIEW

**Comment by mcarrano** [comment]
There seem to be many variations of this component presented in Core that are not yet included here. https://pf-next.netlify.com/components/InputGroup/examples/  Is there a plan to add more variations?

**Comment by boaz1337** [comment]
@mcarrano - sorry, I missed that part. :scream: 
I will add them to the examples tomorrow.

Thanks for noticing.

**Comment by boaz0** [comment]
@dlabaj @tlabaj - yeah I see what you're saying. I added more examples and it's clear to me the style is odd.

Could it be a problem on my side?

Thanks

**Comment by boaz0** [comment]
Thanks @christiemolloy 
I will fix those two problems. :+1:

File: packages/patternfly-4/react-core/src/components/InputGroup/examples/SimpleInputGroups.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  DropdownToggle,
+  Popover
+} from '@patternfly/react-core';
+
+class SimpleInputGroups extends React.Component {
+  render() {
+    return (
+      <React.Fragment>
+        <InputGroup>
+          <Button id="textAreaButton1" variant="secondary">
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/InputGroup/examples/SimpleInputGroups.js]: Can we use variant constant for these?

File: packages/patternfly-4/react-core/src/components/InputGroup/examples/SimpleInputGroups.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+          <TextInput name="textInput11" id="textInput11" type="search" aria-label="search input example" />
+          <Button variant="tertiary" aria-label="search button for search input">
+            <SearchIcon />
+          </Button>
+        </InputGroup>
+        <br />
+        <br />
+        <InputGroup>
+          <TextInput name="textInput10" id="textInput10" type="search" aria-label="input example with popover" />
+          <Popover position="top" bodyContent={'This field is an example of input group with popover'}>
```
Comment (dlabaj) [inline packages/patternfly-4/react-core/src/components/InputGroup/examples/SimpleInputGroups.js]: Same comment as @tlabaj but for position.

**Comment by christiemolloy** [comment]
Only one discrepancy I can see is this example. This is what we have in Core and the button in React is a different color.

screen shot 2019-02-25 at 1 24 01 pm

File: packages/patternfly-4/react-core/src/components/InputGroup/examples/SimpleInputGroups.js (hunk 2 above)
Comment (boaz0) [reply packages/patternfly-4/react-core/src/components/InputGroup/examples/SimpleInputGroups.js]: Good point. Thanks!

File: packages/patternfly-4/react-core/src/components/InputGroup/examples/SimpleInputGroups.js (hunk 1 above)
Comment (boaz0) [reply packages/patternfly-4/react-core/src/components/InputGroup/examples/SimpleInputGroups.js]: Good idea, I will work on that.

**Comment by boaz0** [comment]
@tlabaj can you give me one more day. I believe I can fix this PR according to these changes.
Please :pray:

**Comment by tlabaj** [comment]
@boaz0 @christiemolloy @dlabaj I opened issue #1483 to track the remaining work.

**Comment by boaz0** [comment]
No problem. I am really sorry about it. I was busy with other stuff I forgot about it.
Thank you anyway.

**Comment by boaz0** [comment]
I know what's the problem. Didn't imported `ReactType` from `react`. Writing a fix.

**Pull request merged**